BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/ConfigString/ConfigStringParser.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.ConfigString
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    using static System.FormattableString;
+
+    /// <summary>
+    /// Helper class to parse a configuration string.
+    /// A configuration string is defined as a string composed of key/value pairs.
+    /// (ex: "key1=value1;key2=value2;key3=value3").
+    /// </summary>
+    internal static class ConfigStringParser
+    {
+        private static readonly char[] SplitSemicolon = new char[] { ';' };
+
+        /// <summary>
+        /// A configuration string is defined as a string composed of key/value pairs.
+        /// (ex: "key1=value1;key2=value2;key3=value3").
+        /// </summary>
+        /// <param name="configString">Input string to be parsed. This string cannot be null or empty. This string will be checked for validity.</param>
+        /// <param name="configName">Name of the config string that is being parsed. This is used for error messages.</param>
+        /// <returns>Returns a dictionary of Key/Value pairs. Keys are not case sensitive.</returns>
+        /// <remarks>This is used by both Connection Strings and Self-Diagnostics configuration.</remarks>
+        public static IDictionary<string, string> Parse(string configString, string configName)
+        {
+            if (configString == null)
+            {
+                string message = Invariant($"{configName} cannot be null.");
+                CoreEventSource.Log.ConfigurationStringError(message);
```

Comment:
warning. As this is anyway going to throw and the caller should decide if it should log at Error level.

Reply:
done